usr.sbin/adduser/adduser.sh

```diff
@@ -306,6 +316,11 @@ add_user() {
 		fi
 	fi
 
+	# give newly created user permissions to their home zfs dataset
+	if [ -n "$Zflag" ]; then
+		set_zfs_perms
```

Comment:
This should happen if the user directory is a ZFS dataset (maybe use a similar command like `zfs list` above).